hooks/use-fetch-exr.ts

```diff
@@ -27,13 +32,26 @@ export interface ExchangeRate {
   errorMessage: string | null;
 }
 
+// Exchange rates are only available on weekdays
+//TODO : Missing Bank Holidays
```

Comment:
French government provides an API for this: `https://calendrier.api.gouv.fr/jours-feries/metropole.json` (20 years history and 5 years in the future)
[doc](https://calendrier.api.gouv.fr/jours-feries/)

Reply:
We need US bank holidays though, right?

Reply:
I don't think so, exchange rate is needed to convert USD prices to EUR compute french taxes.

The rate comes from Banque de France, and the problem to address is there is no rate from Banque de France API when this is a weekend or a bank holiday.

Reply:
I just ran this snippet in a browser console:

```
const apiUrl =
  "https://data-api.ecb.europa.eu/service/data/EXR/D.USD.EUR.SP00.A";
const fetchExchangeRate = (date) => {
  const searchParams = new URLSearchParams({
    format: "jsondata",
    detail: "dataonly",
  });
  searchParams.set("startPeriod", date);
  searchParams.set("endPeriod", date);

  return fetch(`${apiUrl}?${searchParams.toString()}`)
    .then((res) => res.json())
    .then(
      (data) =>
        data.dataSets[0].series["0:0:0:0:0"].observations[0][0],
    ).then(r => console.log(r));
};

// Jan 15th is Martin Luther King day
fetchExchangeRate('2024-01-15')
// President day
fetchExchangeRate('2024-02-19')
```

Both returns a value

Reply:
Interesting, I guess if we want to be extra sure we can always have it run for all 365 days of the year and check which ones fail 😅 

If that's the case, which value do we use if the acquisition or sell happened on a French holiday?

Reply:
That's why I went for the simple solutioon, handling Week-ends first :)
My allocation falls on a Saturday
<img width="80" alt="Screenshot 2024-05-10 at 15 28 43" src="https://github.com/hinosxz/tax-helper/assets/478729/a1962286-b601-4eb8-ad10-5f5cb6d61689">

Reply:
We should use the day before AFAICT.
The reason why there is no data on French holidays is that french markets are closed, hence there is no USD/EUR exchange in Paris.
When US markets are off, it does not affect french markets.

We might face the same kind of issue with stock prices, but the whole history is fetched at once, so it is way easier to try to data for the day before until there is some data.